### PR TITLE
Add optional descriptions

### PR DESCRIPTION
**Pull request by tharwan**
This PR adresses #86

### CODE REVIEW

File: tests/test_form.py [hunk 1]
```diff
@@ -25,6 +25,16 @@ def test_form(testcase, preconstruct):
             form.data = doc
 
 
+@pytest.mark.parametrize("preconstruct", (0, 1))
```
Comment (dokempf) [inline tests/test_form.py:28]: No need to test this with this parametrization, as the two features do not interact.

**Review by dokempf — CHANGES_REQUESTED** [review]
This is looking good. My only concern is that it currently only applies to those widgets that went through `_construct_simple` - missing e.g. all formats, object and array. But then again, similar things can be said about tooltips, I guess. If you have the capacity to extend the feature to these additional types, feel free to do so. If you do not have it, I am also fine with merging as is, as it is an improvement over the status quo.

**Comment by tharwan** [comment]
> This is looking good. My only concern is that it currently only applies to those widgets that went through `_construct_simple` - missing e.g. all formats, object and array. But then again, similar things can be said about tooltips, I guess. If you have the capacity to extend the feature to these additional types, feel free to do so. If you do not have it, I am also fine with merging as is, as it is an improvement over the status quo.

I was not sure how to do it for the composed formats, but I also think it is most needed for the actual input fields which all go through the construct_simple.

**Comment by tharwan** [comment]
I would be very happy if we could merge this soon.

**Comment by dokempf** [comment]
@tharwan I implemented the missing types (array, object and datetime formats). With this change, I think I am ready to go.

**Review by dokempf — APPROVED** [review]

**Pull request merged**